src/bam_io.cpp

```diff
@@ -7,7 +7,7 @@
 #include "stringops.h"
 
 void BamAlignment::ExtractSequenceFields(){
-  int32_t length = b_->core.l_qseq;
+  int32_t length = length_; //b_->core.l_qseq; // changed to accomodate end trimming
```

Comment:
can we be sure that length_ has been already instantiated here?

Reply:
I think length_ is set upon BamAlignment object creation to -1 (bam_io.h line 69). b_ is set to whatever bam_init1() returns. I don't think ExtractSequenceFields will fail if you call ExtractSequenceFields on this freshly instantiated object, but I haven't tested this. length_ and the rest of BamAlignment fields should have proper values after GetNextAlignment is called. The reason I changed b_->core.l_qseq to length_ is so that ExtractSequenceFields is agnostic to whether trimming has been done or not. It simply reports the seq, qual and cigar according to the length field of the alignment object itself and doesn't need any extra variables to be passed into it. Not sure why it was b_->core.l_qseq originally since the code extracts information from b_ into fields of the alignment object.